File: app/src/main/java/cn/kgc/tiku/bluebird/adapter/ClassRankingListItemAdapter.java
package cn.kgc.tiku.bluebird.adapter;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

import cn.kgc.tiku.bluebird.R;
import cn.kgc.tiku.bluebird.entity.Ranking;
import cn.kgc.tiku.bluebird.utils.Contant;

/**
 * Created by star on 2018/8/19.
 */

public class ClassRankingListItemAdapter extends ArrayAdapter {
    private final int resourceId;

    public ClassRankingListItemAdapter(Context context, int textViewResourceId, List<Ranking> objects) {
        super(context, textViewResourceId, objects);
        resourceId = textViewResourceId;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        Ranking ranking = (Ranking) getItem(position);
        View view = LayoutInflater.from(getContext()).inflate(resourceId, null);//实例化一个对象
        ((TextView) view.findViewById(R.id.mc)).setText(String.valueOf(ranking.getMc()));
        ((TextView) view.findViewById(R.id.xm)).setText(ranking.getXm());
        ((TextView) view.findViewById(R.id.ljdt)).setText(String.valueOf(ranking.getLjdt()));
        ((TextView) view.findViewById(R.id.sjdt)).setText(String.valueOf(ranking.getSjdt()));
        ((TextView) view.findViewById(R.id.zql)).setText(String.valueOf(ranking.getZql()+"%"));
        if(ranking.getXm().equals(Contant.userInfo.getUserName())){
            view.setBackgroundColor(Color.parseColor("#ff4081"));
        }
        return view;
    }
}

File: app/src/main/java/com/kgc/tiku/bluebird/adapter/ClassRankingListItemAdapter.kt
package com.kgc.tiku.bluebird.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.kgc.tiku.bluebird.R
import com.kgc.tiku.bluebird.entity.Ranking


class ClassRankingListItemAdapter constructor(context: Context, textViewResourceId: Int, objects: List<Ranking>) :
    ArrayAdapter<Ranking>(context, textViewResourceId, objects) {
    private var resourceId: Int = 0

    init {
        resourceId = textViewResourceId
    }

    @SuppressLint("ViewHolder")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val ranking = getItem(position) as Ranking
        val view = LayoutInflater.from(context).inflate(resourceId, null)
        view.findViewById<TextView>(R.id.userName).text = ranking.userName
        view.findViewById<TextView>(R.id.percentage).text = ranking.percentage
        view.findViewById<TextView>(R.id.index).text = ranking.index.toString()
        view.findViewById<TextView>(R.id.answerNo).text = ("累计答题 " + ranking.answerNo)
        view.findViewById<TextView>(R.id.actualQuestionNoNow).text = ("实际答题 " + ranking.actualQuestionNoNow.toString())
        return view
    }

}